File: robot_math/test/visualize.m
fid = fopen("~/ros2_ws/data.txt", "r");
data = fscanf(fid, "%f", [4, inf]);
fclose(fid);
plot(data(1,:), data(2, :), 'r', 'LineWidth', 2);
hold on;
plot(data(1,:), data(3, :), 'b', 'LineWidth', 2);
plot(data(1,:), data(4, :), 'g', 'LineWidth', 2);
hold off;

File: robot_math/test/visualize.py
import matplotlib.pyplot as plt
import numpy as np

data = np.loadtxt('data.txt')
t = data[:,0]
x = data[:,1]
y = data[:,2]
vx = data[:,3]
vy = data[:,4]
ax = data[:,5]
ay = data[:,6]
plt.plot(t,x,t,y,t,vx,t,vy)
plt.show()